daemon/container_linux.go

```diff
@@ -306,6 +307,24 @@ func populateCommand(c *Container, env []string) error {
 	return nil
 }
 
+func mergeDevices(defaultDevices, userSpecifiedDevices []*configs.Device) []*configs.Device {
+	if len(userSpecifiedDevices) == 0 {
+		return defaultDevices
+	}
+
+	// Only add those default devices, which are not specified by the user.
+	// Otherwise, the user specified devices end up without permissions.
+	var devs []*configs.Device
+	for _, defaultDevice := range defaultDevices {
+		for _, userSpecified := range userSpecifiedDevices {
+			if defaultDevice.Path != userSpecified.Path {
```

Comment:
I understand what you are trying to do, but I don't get how your code does this. My python adled brain is reading this as:

````
for defaultDevice in defaultDevices:
  for userSpecified in userSpecifiedDevices:
    if defaultDevice.Path != userSpecified.Path:
      devs = append(devs, defaultDevice)
````

Which would be wrong.